lib/Sources/JsMessage/JsLogger.swift

```diff
@@ -17,8 +25,82 @@ public final class JsLogger: JsMessageProcessor {
         guard let message = request.body["msg"]?.string else {
             return .error(DomError(name: .encoding, message: "Log body missing required field 'msg'"))
         }
-        // For now just use the console for debug logging
-        print("JsLog: \(message)")
+        let level = decodeLevel(request.body["level"]?.string)
+        let sensitive = request.body["sensitive"]?.number?.boolValue ?? false
+        let isConsole = request.body["console"]?.number?.boolValue ?? false
+        let data = request.body["data"]?.dictionary ?? [:]
+        sendLog(level: level, isConsole: isConsole, sensitive: sensitive, message: message, data: data)
         return .body([:])
     }
+
+    private func sendLog(level: Level, isConsole: Bool, sensitive: Bool, message: String, data: [String: JsType]?) {
+        if isConsole {
+            sendLogConsole(level: level, message: message, data: JsType.dictionaryAsString(data))
+        } else if sensitive {
+            sendLogRedacted(level: level, message: message, data: JsType.dictionaryAsString(data))
+        } else {
+            sendLogPublic(level: level, message: message, data: JsType.dictionaryAsString(data))
+        }
+    }
+
+    private func sendLogConsole(level: Level, message: String, data: String) {
+        switch level {
+        case .debug:
+            consoleLog.debug("\(message, privacy: .public) \(data, privacy: .public)")
+        case .info:
+            consoleLog.info("\(message, privacy: .public) \(data, privacy: .public)")
+        case .warn:
+            consoleLog.warning("\(message, privacy: .public) \(data, privacy: .public)")
+        case .error:
+            consoleLog.error("\(message, privacy: .public) \(data, privacy: .public)")
+        }
+    }
+
+    private func sendLogPublic(level: Level, message: String, data: String) {
+        switch level {
+        case .debug:
+            log.debug("\(message, privacy: .public) \(data, privacy: .public)")
+        case .info:
+            log.info("\(message, privacy: .public) \(data, privacy: .public)")
+        case .warn:
+            log.warning("\(message, privacy: .public) \(data, privacy: .public)")
+        case .error:
+            log.error("\(message, privacy: .public) \(data, privacy: .public)")
+        }
+    }
+
+    private func sendLogRedacted(level: Level, message: String, data: String) {
```

Comment:
Note that private will be redacted from the system logs, but will still show up if you have a debugger atatched.

Reply:
Maybe we should add a comment for that?

Reply:
It is well documented in the official OSLog documentation so doesn't seem necessary. I called it out here for reviewers who may not be familiar with the system.